### ISSUE TITLE
Issues with SDK 9.0.0GA Android build

### ISSUE THREAD

**Issue by macasfaj** [Pushwoosh/pushwoosh-appcelerator-titanium#75]
Hi, I've just updated my App to SDK 9.0.0GA and I can't compile on Android.

I get this issue when building:

[ERROR]  Error: ENOENT: no such file or directory, open '/Users/Migue/Documents/NubecitaSVN/Supernova/build/android/app/src/main/AndroidManifest.xml'

at Object.openSync (fs.js:462:3)

at Object.readFileSync (fs.js:364:35)

at AndroidBuilder.post (/Users/Migue/Documents/NubecitaSVN/Supern/modules/android/com.pushwoosh.module/5.20.0/hooks/gen_google_services_config.js:26:26)

at /Users/Migue/.appcelerator/install/8.0.0/package/node_modules/titanium/lib/hook.js:246:13

at /Users/Migue/.appcelerator/install/8.0.0/package/node_modules/titanium/node_modules/async/dist/async.js:3880:24

at replenish (/Users/Migue/.appcelerator/install/8.0.0/package/node_modules/titanium/node_modules/async/dist/async.js:1011:17)

at /Users/Migue/.appcelerator/install/8.0.0/package/node_modules/titanium/node_modules/async/dist/async.js:1016:9

at eachOfLimit (/Users/Migue/.appcelerator/install/8.0.0/package/node_modules/titanium/node_modules/async/dist/async.js:1041:24)

at /Users/Migue/.appcelerator/install/8.0.0/package/node_modules/titanium/node_modules/async/dist/async.js:1046:16

at _parallel (/Users/Migue/.appcelerator/install/8.0.0/package/node_modules/titanium/node_modules/async/dist/async.js:3879:5

Could you check, please? Thanks.

**Comment by wfhm** [comment]
Hi @macasfaj,

Checking it.

**Comment by wfhm** [comment]
@macasfaj,

Starting with 9.0.0 GA, projects are now built with gradle instead of ant. This significant change affected the order of calling our hooks that were used to modify AndroidManifest.xml.

So, after gradle was introduced, the **google-services** gradle plugin cannot be automatically added to build.gradle, and google-services.json cannot be placed in **android** directory neither with a hook nor with any other tool. The only option right now is to manually edit build.gradle on your side after native Android project is built.

Appcelerator team already works on fixing this (you can find the details here - https://jira.appcelerator.org/browse/TIMOB-27745), so we are now waiting for them to release 9.0.1 version before we update our hooks.

**Comment by macasfaj** [comment]
Ok. I’ll wait. Thanks 😊

**Comment by arifje** [comment]
Hi, 9.0.1.GA is released last week, any update on this issue? Tested the current version of the module (5.20.0), but it's not compiling.

**Comment by wfhm** [comment]
@skoften, @macasfaj,

After Titanium fixed the issue on their side, we still have to update the plugin to make it work on our side. Right now team is finishing works on the 6.0.0 SDK release, so the plugins will be updated after the major release is rolled out. I will provide ETA as soon as there is any.

**Comment by macasfaj** [comment]
Good morning. Any update about this? Thanks!

**Comment by wfhm** [comment]
@macasfaj,

The Titanium plugin was not yet updated, we are still working on it. I will provide an updated ETA shortly.

**Comment by macasfaj** [comment]
Good morning. Sorry for being pushy. Any update? We need to move to SDK 9 ASAP. Thanks!

**Comment by arifje** [comment]
Same here!!

**Comment by phobeous** [comment]
We're also looking forward to upgrade Ti SDK, but cannot because of this issue. Any forecast for a release??

**Comment by jfalcone** [comment]
There are many alternatives to Pushwoosh now and we're going to start investigating migration to another notification service.  This is a serious dropping of the ball by pushwoosh.  No business can tolerate 7 months of not being able to upgrade to new functionality because of one component.

**Comment by wfhm** [comment]
@jfalcone,

You are right. We have adjusted our roadmap and the fix to this issue will be rolled out without updating native SDKs to 6.+ versions shortly.

**Comment by macasfaj** [comment]
Great news. Thanks!

**Comment by wfhm** [comment]
Hey @macasfaj, @jfalcone, @phobeous, @skoften

The new version of the module is released, the new titanium version is supported with it.

**Please note** that the plugin still manually adds jars and aars, and, as Titanium started to use gradle with 9.0.0 version, there might be dependency conflicts. These conflicts will be fixed as soon as we release the 6.0 version of the plugin. We will provide the ETA shortly.

**Comment by arifje** [comment]
Any update on the 6.0 version? We are having conflicts with ti.android.admob module. 

Tried removing firebase-* jars, but can't get it to work, keep getting this error:

```
09-04 19:44:44.436 10629 10629 E TiExceptionHandler: (main) [212,495] Unable to instantiate service com.pushwoosh.PushFcmIntentService: java.lang.ClassNotFoundException: Didn't find class "com.pushwoosh.PushFcmIntentService" on path: DexPathList[[zip file "/data/app/xxx.android.app-1/base.apk"],nativeLibraryDirectories=[/data/app/xxx.android.app-1/lib/arm64, /data/app/xxx.android.app-1/base.apk!/lib/arm64-v8a, /system/lib64, /vendor/lib64]]
09-04 19:44:44.436 10629 10629 E TiExceptionHandler: 
09-04 19:44:44.436 10629 10629 E TiExceptionHandler:     dalvik.system.BaseDexClassLoader.findClass(BaseDexClassLoader.java:56)
09-04 19:44:44.436 10629 10629 E TiExceptionHandler:     java.lang.ClassLoader.loadClass(ClassLoader.java:380)
09-04 19:44:44.436 10629 10629 E TiExceptionHandler:     java.lang.ClassLoader.loadClass(ClassLoader.java:312)
09-04 19:44:44.436 10629 10629 E TiExceptionHandler:     android.app.ActivityThread.handleCreateService(ActivityThread.java:3446)
09-04 19:44:44.436 10629 10629 E TiExceptionHandler:     android.app.ActivityThread.-wrap6(ActivityThread.java)
09-04 19:44:44.436 10629 10629 E TiExceptionHandler:     android.app.ActivityThread$H.handleMessage(ActivityThread.java:1721)
09-04 19:44:44.436 10629 10629 E TiExceptionHandler:     android.os.Handler.dispatchMessage(Handler.java:102)
09-04 19:44:44.436 10629 10629 E TiExceptionHandler:     android.os.Looper.loop(Looper.java:154)
09-04 19:44:44.436 10629 10629 E TiExceptionHandler:     android.app.ActivityThread.main(ActivityThread.java:6682)
09-04 19:44:44.436 10629 10629 E TiExceptionHandler:     java.lang.reflect.Method.invoke(Native Method)
09-04 19:44:44.436 10629 10629 E TiExceptionHandler:     com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1520)
09-04 19:44:44.436 10629 10629 E TiExceptionHandler:     com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1410)
09-04 19:44:45.020 10799 10799 E Zygote  : v2
09-04 19:44:45.022 10799 10799 E Zygote  : accessInfo : 0
```

And then the app crashes.

Thanks!

**Comment by macasfaj** [comment]
Same here, I need 6.0 :/

**Comment by macasfaj** [comment]
Why is this issue closed? I have issues with other modules using SDK 9

**Comment by wfhm** [comment]
@skoften  @macasfaj We have already started working on the 6.0.0 update, the ETA is the end of September.

**Comment by wfhm** [comment]
@macasfaj The original issue was related to Titanium hooks not working with 9.0.0 GA, which was fixed with 5.20.1, hence it was closed. The new build issue you have should be discussed in a separate thread [here](https://github.com/Pushwoosh/pushwoosh-appcelerator-titanium/issues/77).

**Comment by macasfaj** [comment]
ok, thanks! :)
